src/qt/walletmodel.cpp

```diff
@@ -926,27 +927,25 @@ int64_t WalletModel::getKeyCreationTime(const libzcash::SaplingPaymentAddress& a
     return 0;
 }
 
-PairResult WalletModel::getNewAddress(Destination& ret, std::string label) const
+CallResult<Destination> WalletModel::getNewAddress(const std::string& label) const
 {
-    CTxDestination dest;
-    PairResult res = wallet->getNewAddress(dest, label);
-    if (res.result) ret = Destination(dest, false);
-    return res;
+    auto res = wallet->getNewAddress(label);
+    return res ? CallResult<Destination>(Destination(*res.getObjResult(), false)) :
+           CallResult<Destination>(res.getError());
 }
 
-PairResult WalletModel::getNewStakingAddress(Destination& ret,std::string label) const
+CallResult<Destination> WalletModel::getNewStakingAddress(const std::string& label) const
 {
-    CTxDestination dest;
-    PairResult res = wallet->getNewStakingAddress(dest, label);
-    if (res.result) ret = Destination(dest, true);
-    return res;
+    auto res = wallet->getNewStakingAddress(label);
+    return res ? CallResult<Destination>(Destination(*res.getObjResult(), true)) :
+           CallResult<Destination>(res.getError());
 }
 
-PairResult WalletModel::getNewShieldedAddress(QString& shieldedAddrRet, std::string strLabel)
+CallResult<Destination> WalletModel::getNewShieldedAddress(QString& shieldedAddrRet, std::string strLabel)
 {
     shieldedAddrRet = QString::fromStdString(
             KeyIO::EncodePaymentAddress(wallet->GenerateNewSaplingZKey(strLabel)));
-    return PairResult(true);
+    return CallResult<Destination>(true, "");
 }
```

Comment:
Agree on the ambiguity of this.

Have a good option C to standardize and clean the functionality:
The `Destination` class is an addr IO wrapper (essentially abstracts the fact that there are different key types and addr encodings to the upper layers). So, instead of store a `CTxDestination`, it can store a `CWDestination`(which is a variant of `CTxDestination` and `SaplingPaymentAddress`), then the rest connects with this quite easily, the destination_io `Standard` namespace already has general `EncodeDestination` and `isValidDestination` functions for every key and address type.

(Pushed it in 9f03042 and 5e6339d).

Reply:
Sure. The point here was about returning the value in one single place (either `CallResult` or a reference argument), not both.
All good now :+1: